packages/ui-patterns/AssistantChat/AssistantChatForm.tsx

```diff
@@ -0,0 +1,148 @@
+import { Loader2 } from 'lucide-react'
+import React, { ChangeEvent, createRef, useEffect } from 'react'
+import { TextArea } from 'ui/src/components/shadcn/ui/text-area'
+import { cn } from 'ui/src/lib/utils'
+
+export interface FormProps extends React.FormHTMLAttributes<HTMLFormElement> {
+  /* The ref for the textarea */
+  textAreaRef: React.RefObject<HTMLTextAreaElement>
+  /* The loading state of the form */
+  loading: boolean
+  /* The disabled state of the form */
+  disabled?: boolean
+  /* The value of the textarea */
+  value?: string
+  /* The function to handle the value change */
+  onValueChange: (value: ChangeEvent<HTMLTextAreaElement>) => void
+  /* Used to stop onSubmit event when Command popover is open. Use with AssistantCommandsPopover */
+  commandsOpen?: boolean
+  /* Used to close Command popover when onSubmit event happens. Use with AssistantCommandsPopover */
+  setCommandsOpen?: (value: boolean) => void
+  /* The icon to display on the left of the textarea */
+  icon?: React.ReactNode
+  /* The function to handle the form submission */
+  onSubmit: React.FormHTMLAttributes<HTMLFormElement>['onSubmit']
+}
+
+const AssistantChatForm = React.forwardRef<HTMLFormElement, FormProps>(
+  (
+    {
+      loading = false,
+      disabled = false,
+      value = '',
+      textAreaRef,
+      commandsOpen = false,
+      icon = null,
+      onValueChange,
+      setCommandsOpen,
+      onSubmit,
+      ...props
+    },
+    ref
+  ) => {
+    const formRef = createRef<HTMLFormElement>()
+    const submitRef = createRef<HTMLButtonElement>()
+
+    /**
+     * This effect is used to resize the textarea based on the content
+     */
+    useEffect(() => {
+      if (textAreaRef) {
+        if (!value && textAreaRef && textAreaRef.current) {
+          textAreaRef.current.style.height = '40px'
+        } else if (textAreaRef && textAreaRef.current) {
+          const newHeight = textAreaRef.current.scrollHeight + 'px'
+          textAreaRef.current.style.height = newHeight
+        }
+      }
+    }, [value, textAreaRef])
+
+    /**
+     * This effect is used to focus the textarea when the component mounts
+     */
+    useEffect(() => {
+      textAreaRef?.current?.focus()
+    }, [value, textAreaRef])
```

Comment:
Is `value` needed in the dependency array here? It'll mean it calls `focus()` on every keystroke

```suggestion
    }, [textAreaRef])
```

Reply:
ah, i think it is. maybe i should add some comments. it's so weird how it works.

I think it's because the value changes externally from the CommandPopover, you need to refocus it again to the input

Reply:
i feel like this useEffect shouldn't be in here at all, and the app can handle this instead (if it needs to)